### DISCUSSION TITLE
Casting elements of an array

### DISCUSSION THREAD

**Discussion by tborealis** [spatie/laravel-data#614] [Q&A] [answered]
I'm attempting to cast individual elements of an array but not having much success. The result of the cast does not implement `Spatie\LaravelData\Data`. This means I can't use the DataCollectionOf attribute and the Cast gets passed the whole array. I'd like it to be passed individual elements to reduce the scope of the cast. Any pointers on how to implement this?

**Comment by rubenvanassche** [comment, marked as answer]
Write a custom cast which takes in the whole array and then casts the individual elements, there isn't another way to solve this as far as I'm aware of.

**Reply by tborealis** [reply]
This is the approach I ended up taking. Has the situation changed with the release of v4?

**Reply by rubenvanassche** [reply]
Yeah we've merged iterableItem casts a few days ago, that should normally solve your problem.
